modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java

```diff
@@ -2040,6 +2040,10 @@ protected C createConsumerImpl(final Destination destination, final int prefetch
                                                  final int prefetchLow, final boolean noLocal, final boolean exclusive, String selector, final FieldTable rawSelector,
                                                  final boolean noConsume, final boolean autoClose) throws JMSException
     {
+        if (this._consumers.values().size() > 0 ) {
+            throw new JMSException("Only one consumer can be created per session");
```

Comment:
Should we do this in the client code? Since the client should behave according the JMS spec and since we cannot change other client implementations (RabittMQ etc.) I think we should not do this? May be we can throw and error in the back-end or actually fix it (I don't know how much time it will take :smile:).

WDYT?